src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +46,391 @@ ActorInit En_Takaraya_InitVars = {
+void func_80ADF7CC(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF520.s")
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        if (this->actor.textId == 0x77A) {
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED)) {
+                Message_CloseTextbox(play);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+                EnTakaraya_SetupWait(this);
+            } else {
+                this->actor.textId = 0x77C;
+                Message_ContinueTextbox(play, this->actor.textId);
+            }
+        } else {
+            this->actor.textId = D_80ADFB50[GET_PLAYER_FORM];
+            Message_ContinueTextbox(play, this->actor.textId);
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+            EnTakaraya_SetupTalk(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF608.s")
+void EnTakaraya_Update(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+    s32 pad;
+    Vec3s torsoRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF654.s")
+    this->actionFunc(this, play);
+    Actor_TrackPlayer(play, &this->actor, &this->headRot, &torsoRot, this->actor.focus.pos);
+    EnTakaraya_Blink(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF6DC.s")
+s32 EnTakaraya_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnTakaraya* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF730.s")
+    if (limbIndex == TREASURE_CHEST_SHOP_GAL_LIMB_HEAD) {
+        rot->x += this->headRot.y;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF7B8.s")
+void EnTakaraya_TransformLimbDraw(PlayState* play, s32 limbIndex, Actor* thisx) {
+    EnTakaraya* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF7CC.s")
+    if (limbIndex == TREASURE_CHEST_SHOP_GAL_LIMB_HEAD) {
+        Matrix_RotateYS(0x400 - this->headRot.x, MTXMODE_APPLY);
+        Matrix_MultVecX(500.0f, &thisx->focus.pos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Update.s")
+void EnTakaraya_Draw(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF94C.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    Gfx_SetupDL25_Opa(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF984.s")
+    if ((gSaveContext.save.playerForm == PLAYER_FORM_DEKU) || (gSaveContext.save.playerForm == PLAYER_FORM_HUMAN)) {
```

Comment:
```suggestion
    if ((GET_PLAYER_FORM == PLAYER_FORM_DEKU) || (GET_PLAYER_FORM == PLAYER_FORM_HUMAN)) {
```
Same here, does this still match?